### ISSUE TITLE
PydanticUserError

### ISSUE THREAD

**Issue by SergeyPriem** [streamsync-cloud/streamsync#51]
After installation of streamsync through **pip install "streamsync[ds]"** and running **streamsync hello** PydanticUserError is raised:   

File "C:\term\venv\Lib\site-packages\pydantic\_internal\_generate_schema.py", line 800, in _typed_dict_schema
    raise PydanticUserError(
pydantic.errors.PydanticUserError: Please use `typing_extensions.TypedDict` instead of `typing.TypedDict` on Python < 3.12.

(I'm using Python 3.11)

**Comment by martinmarroyo** [comment]
I ran into the same issue when installing streamsync 0.1.9 and running `streamsync hello` on Python 3.11. Here is the relevant documentation from Pydantic: https://docs.pydantic.dev/dev-v2/usage/errors/#typed-dict-version. This is breaking the `streamsync` command in general on my end. I modified the imports in `ss_types.py` to check for the python version and import TypedDict based on that. It got the `streamsync` command to work, but now I'm getting validation errors when running `streamsync hello`.

I'd like to help with troubleshooting this further, but am admittedly a little lost with respect to collaboration on this project and how to build to test my changes. @ramedina86 I'm happy to help, please let me know if that's something you would like.

**Comment by ramedina86** [comment]
I've released version 0.1.10 which pins Pydantic to version 1.10.11 and thus solves the issue.

@SergeyPriem Thanks for reporting this

@martinmarroyo I really appreciate your help on this. As I wrote above, I decided to pin the Pydantic version for now.

The code you wrote seems to address the demands of Pydantic 2, but, as you noted, problems remain (an exception coming from the ASGI application).

I believe that this is a FastAPI issue. Pydantic v2 support was added in the latest version of FastAPI (0.100) and there appears to be some growing pains judging by discussions / issues on their repository. Thankfully, FastAPI decided to keep backwards compatibility with Pydantic v1 which allows to use the latest FastAPI while staying on Pydantic v1.
